applications/ConvectionDiffusionApplication/python_scripts/response_functions/convection_diffusion_response.py

```diff
@@ -0,0 +1,181 @@
+"""
+This module contains an interface to the available response functions
+"""
+import time as timer
+
+import KratosMultiphysics as KM
+from KratosMultiphysics import Parameters, Logger
+import KratosMultiphysics.ConvectionDiffusionApplication as ConvectionDiffusionApplication
+from KratosMultiphysics.ConvectionDiffusionApplication.convection_diffusion_analysis import ConvectionDiffusionAnalysis
+
+
+class ResponseFunctionBase(object):
```

Comment:
isnt there a baseclass in the core?

Reply:
No

Reply:
The naming here is prob. a little bit confusing:

There is a base class for the cpp implementation of an adjoint response function as it is used in the adjoint analysis. However that is a very low level class. For shape optimization we need a top level interface (we don't care if adjoint or whatever other method is used internally).

Reply:
But is this the same one as in structural then? And what abt FluidDynamics and PotentialFlow?
I am just wondering of copy-paste could be avoided if you have a common base class in the Core

Reply:
Yes, this is the same interface. And yes copy paste could be avoided if it was in the core. 
I am just no sure if it fits in to the core.

Reply:
ok you know better, but to me looks like sth for the core :)

You could make a draft PR to see how much code is actually duplicated, then it could be discussed there

Reply:
If these classes are intended to be always implemented in Python, maybe makes sense to create an auxiliary `AdjointWatheverApplication`.

In any case, I agree with @philbucher. We can create an `adjoint_stuff` folder in the KratosCore.